randovania/games/samus_returns/gui/preset_settings/msr_patches_tab.py

```diff
@@ -1,37 +1,37 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+import typing
 
 from PySide6 import QtWidgets
 
 from randovania.games.samus_returns.layout import MSRConfiguration
+from randovania.gui.generated.preset_msr_patches_ui import Ui_PresetMSRPatches
 from randovania.gui.preset_settings.preset_tab import PresetTab
 
-if TYPE_CHECKING:
+if typing.TYPE_CHECKING:
     from randovania.game_description.game_description import GameDescription
     from randovania.gui.lib.window_manager import WindowManager
     from randovania.interface_common.preset_editor import PresetEditor
     from randovania.layout.preset import Preset
 
+_FIELDS = [
+    "elevator_grapple_blocks",
+    "area3_interior_shortcut_no_grapple",
+    "nerf_power_bombs",
+    "nerf_super_missiles",
+]
 
-class PresetMSRPatches(PresetTab):
+
+class PresetMSRPatches(PresetTab, Ui_PresetMSRPatches):
     def __init__(self, editor: PresetEditor, game_description: GameDescription, window_manager: WindowManager):
         super().__init__(editor, game_description, window_manager)
+        self.setupUi(self)
 
-        self.root_widget = QtWidgets.QWidget(self)
-        self.root_layout = QtWidgets.QVBoxLayout(self.root_widget)
-
-        self.include_extra_pickups_check = QtWidgets.QCheckBox(self.root_widget)
-        self.include_extra_pickups_check.setEnabled(True)
-        self.include_extra_pickups_check.setText("Include Extra Pickups")
-        self.root_layout.addWidget(self.include_extra_pickups_check)
-
-        self.include_extra_pickups_label = QtWidgets.QLabel(self.root_widget)
-        self.include_extra_pickups_label.setWordWrap(True)
-        self.include_extra_pickups_label.setText("Include some optional pickups.")
-        self.root_layout.addWidget(self.include_extra_pickups_label)
+        self.setCentralWidget(self.centralWidget)
```

Comment:
Besides the mypy error (which is a lie?), what does it do?
Setting the central widget to the `centralWidget` looks redudant. I think other parts of code in randovania setting it to a different / newly created widget.

Reply:
The widget in the ui is called `centralWidget`. I assume `SetCentralWidget` is just what the function is called.

Reply:
it's a very weird thing. `centralWidget()` is a method of QWidget, but a lot of the ui files have the root widget being named `centralWidget`, which shadows the method. It is a valid mypy error indeed

Reply:
So world the error be fixed if the widget was renamed to something else?

Reply:
I usually name it `root_widget` or something in Qt Creator

Reply:
> it's a very weird thing. `centralWidget()` is a method of QWidget

Yeah, I knew that but I didn't know that we define it in the ui file  and shadow it. We really should avoid that.